Firmware/ultralcd.cpp

```diff
@@ -5695,18 +5680,21 @@ void lcd_thermal_model_cal()
 
 void lcd_sdcard_stop()
 {
-    // Show static message
-    lcd_puts_at_P(0, 0, _T(MSG_STOP_PRINT));
-    lcd_putc_at(0, 1, '\n');
-
-    MENU_BEGIN();
-    // Skip first two LCD rows used by static message
-    if(menu_row == 0) menu_row = 2;
+uint8_t result = lcd_show_fullscreen_message_yes_no_and_wait_P(_T(MSG_STOP_PRINT), true, LCD_MIDDLE_BUTTON_CHOICE);
```

Comment:
This change is not ok. It might save resources since it reuses code, but it breaks one of the quirks of the old implementation: gcodes are no longer queued while the message is displayed. This is in contrast to the old implementation which behaved like a menu that gets periodically re-rendered by lcd_update. As a result, with this change when you click stop print, the printer just drains all the queued moves in the planner queue and then the planner stalls since gcode are not processed during lcd_show_fullscreen_message_yes_no_and_wait_P

Reply:
I agree with @leptun here. This menu must be “non-blocking” so it doesnt halt the print.